//pkg/kubelet/pleg:go_default_test is flaky

@kubernetes/sig-node-bugs 

```
FAIL: //pkg/kubelet/pleg:go_default_test (see /usr/local/google/home/fejta/.cache/bazel/_bazel_fejta/631ceaac9e4091781154c1030ef08d3e/execroot/__main__/bazel-out/k8-fastbuild/testlogs/pkg/kubelet/pleg/go_default_test/test_attempts/attempt_1.log)

FLAKY: //pkg/kubelet/pleg:go_default_test (Summary)
      /usr/local/google/home/fejta/.cache/bazel/_bazel_fejta/631ceaac9e4091781154c1030ef08d3e/execroot/__main__/bazel-out/k8-fastbuild/testlogs/pkg/kubelet/pleg/go_default_test/test.log
      /usr/local/google/home/fejta/.cache/bazel/_bazel_fejta/631ceaac9e4091781154c1030ef08d3e/execroot/__main__/bazel-out/k8-fastbuild/testlogs/pkg/kubelet/pleg/go_default_test/test_attempts/attempt_1.log
INFO: From Testing //pkg/kubelet/pleg:go_default_test:
==================== Test output for //pkg/kubelet/pleg:go_default_test:
E0424 06:42:20.674764      18 generic.go:274] event channel is full, discard this relist() cycle event
--- FAIL: TestEventChannelFull (0.00s)
    generic_test.go:97: Actual events differ from the expected; diff:
         [{"ID":"1234","Type":"ContainerRemoved","Data":"c1"},{"ID":"1234","Type":"ContainerDied","Data":"c2"},{"ID":"
        
        A: 1234","Type":"ContainerStarted","Data":"c3"},{"ID":"4567","Type":"ContainerRemoved","Data":"c1"}]
        
        B: 4567","Type":"ContainerRemoved","Data":"c1"},{"ID":"4567","Type":"ContainerStarted","Data":"c4"}]
        
FAIL
```

--- a/pkg/kubelet/pleg/generic_test.go
+++ b/pkg/kubelet/pleg/generic_test.go
@@ -217,9 +217,17 @@ func TestEventChannelFull(t *testing.T) {
 		{ID: "1234", Type: ContainerDied, Data: "c2"},
 		{ID: "1234", Type: ContainerStarted, Data: "c3"},
 		{ID: "4567", Type: ContainerRemoved, Data: "c1"},
+		{ID: "4567", Type: ContainerStarted, Data: "c4"},
 	}
 	actual = getEventsFromChannel(ch)
-	verifyEvents(t, expected, actual)
+	assert.True(t, len(actual) == 4, "channel length should be 4")
+	for _, actualEvent := range actual {
+		for _, expectedEvent := range expected {
+			if actualEvent.ID == expectedEvent.ID && actualEvent.Data == expectedEvent.Data {
+				assert.True(t, actualEvent.Type == expectedEvent.Type, "event type should be equal")
+			}
+		}
+	}
 }
 
 func TestDetectingContainerDeaths(t *testing.T) {

--- a/pkg/kubelet/pleg/generic_test.go
+++ b/pkg/kubelet/pleg/generic_test.go
@@ -211,23 +211,17 @@ func TestEventChannelFull(t *testing.T) {
 		}},
 	}
 	pleg.relist()
-	// event channel is full, discard events
-	expected = []*PodLifecycleEvent{
+	allEvents := []*PodLifecycleEvent{
 		{ID: "1234", Type: ContainerRemoved, Data: "c1"},
 		{ID: "1234", Type: ContainerDied, Data: "c2"},
 		{ID: "1234", Type: ContainerStarted, Data: "c3"},
 		{ID: "4567", Type: ContainerRemoved, Data: "c1"},
 		{ID: "4567", Type: ContainerStarted, Data: "c4"},
 	}
+	// event channel is full, discard events
 	actual = getEventsFromChannel(ch)
 	assert.True(t, len(actual) == 4, "channel length should be 4")
-	for _, actualEvent := range actual {
-		for _, expectedEvent := range expected {
-			if actualEvent.ID == expectedEvent.ID && actualEvent.Data == expectedEvent.Data {
-				assert.True(t, actualEvent.Type == expectedEvent.Type, "event type should be equal")
-			}
-		}
-	}
+	assert.Subsetf(t, allEvents, actual, "actual events should in all events")
 }
 
 func TestDetectingContainerDeaths(t *testing.T) {

